cfy_manager/main.py

```diff
@@ -859,7 +872,44 @@ def configure(verbose=False,
 
     config[UNCONFIGURED_INSTALL] = False
     logger.notice('Configuration finished successfully!')
-    _finish_configuration()
+    _finish_configuration(components)
+
+
+def _all_services_removed(dir_path):
+    return all(
+        not os.path.exists(os.path.join(dir_path, service_name))
+        for service_name in [DATABASE_SERVICE, QUEUE_SERVICE, MANAGER_SERVICE])
+
+
+def _remove_services_from_initial_files(dir_path):
+    for installed_service in get_installed_services_names():
+        _remove(os.path.join(dir_path, installed_service))
+        if dir_path == INITIAL_INSTALL_DIR:
+            update_yaml_file(INSTALLED_COPONENTS, {installed_service: []})
```

Comment:
Why do we need to keep the yaml dict as well as the individual files?

Reply:
That's a great question. I thought about keeping the dictionary in the service-names' files, but then we would have had 2 options: 
1. Changing the name of the file from "<service-name>" to "<service-name>.yaml".
2. Keeping the same file name (i.e. "<service-name>") and treating it as a yaml file. 

I didn't like both of these options, so my idea was that we would have the service names' files in order to figure out which services are installed, and the components.yaml and packages.yaml to figure out which components/packages are installed.

Reply:
Right, but what's the point keeping the dictionary at all? Do we actually use it anywhere? I feel like I'm missing something.

Reply:
Sorry for not explaining this earlier. The purpose of this whole mechanism is to know which components and packages need to be removed in the case of a 3 nodes cluster configuration. The usage of the different dictionaries can be found here: https://github.com/cloudify-cosmo/cloudify-manager-install/blob/d1e0d8bd8ea6ce037ddd7fc4390b40ca405977ab/cfy_manager/main.py#L955-L962.